### PR TITLE
Wrong Gravatar on Messaging

### LINKED ISSUE #1454: Wrong Gravatar on Messaging
**Describe the bug**
When you select "Identicon" has default (gr)avatar, the avatar shown on the messaging is the default gravatar icon and not an "identicon".

image

But the avatar of Luna is the following : 

image

**To Reproduce**
1. Go to wp-admin and change the default avatar (select Identicon)
2. Go to your inbox

See the avatar.

**Expected behavior**

**Screenshots**

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.

### PR DESCRIPTION
**Pull request by dipskakadiya**
Wrong Gravatar on Messaging Issue resolve

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### How to test the changes in this Pull Request:

1. Go to wp-admin and change the default avatar (select Identicon)
2. Go to your inbox

### Proof Screenshots or Video

Screenshot 2020-09-08 at 12 51 32 AM

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php [hunk 1]
```diff
@@ -25,23 +25,23 @@
                 if( data.avatars.length == 2 ) { #>
                     <div class="thread-multiple-avatar">
                 <# } #>
-                    <img class="avatar" src="{{data.avatars[0].url}}" alt="{{data.avatars[0].name}}"/>
+                    <img class="avatar 1" src="{{{data.avatars[0].url}}}" alt="{{data.avatars[0].name}}"/>
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php:28]:
```suggestion
                    <img class="avatar" src="{{{data.avatars[0].url}}}" alt="{{data.avatars[0].name}}"/>
```
I think `1` is for your debug purpose you added so need to remove.

File: src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php [hunk 2]
```diff
@@ -25,23 +25,23 @@
...
                     <# if( data.avatars[1] ) { #>
-                        <img class="avatar" src="{{data.avatars[1].url}}" alt="{{data.avatars[1].name}}"/>
+                        <img class="avatar 2" src="{{{data.avatars[1].url}}}" alt="{{data.avatars[1].name}}"/>
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php:30]:
```suggestion
                        <img class="avatar 2" src="{{{data.avatars[1].url}}}" alt="{{data.avatars[1].name}}"/>
```
I think `2` is for your debug purpose you added so need to remove.

File: src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php [hunk 3]
```diff
@@ -25,23 +25,23 @@
...
                     <# }
                 if( data.avatars.length == 2 ) { #>
                     </div>
                 <# } #>
             <# } else if ( data.group_avatar && data.group_avatar.length > 1 && data.is_group_thread ) { #>
-				<img class="avatar" src="{{data.group_avatar}}" alt="{{data.group_name}}" />
+				<img class="avatar 3" src="{{{data.group_avatar}}}" alt="{{data.group_name}}" />
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php:36]:
```suggestion
				<img class="avatar" src="{{{data.group_avatar}}}" alt="{{data.group_name}}" />
```

File: src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php [hunk 4]
```diff
@@ -25,23 +25,23 @@
...
 			<# } else { #>
 				<# if ( other_recipients.length > 1 ) { #>
 					<span class="recipients-count">{{other_recipients.length}}</span>
-					<img class="avatar" src="{{data.sender_avatar}}" alt="{{data.sender_name}}" />
+					<img class="avatar 4" src="{{{data.sender_avatar}}}" alt="{{data.sender_name}}" />
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php:40]:
```suggestion
					<img class="avatar" src="{{{data.sender_avatar}}}" alt="{{data.sender_name}}" />
```

File: src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php [hunk 5]
```diff
@@ -25,23 +25,23 @@
...
 				<# } else { #>
 					<# var recipient = _.first(other_recipients)? _.first(other_recipients) : current_user; #>
 					<# if ( typeof( recipient ) != "undefined" && recipient !== null && recipient.avatar.length > 1 && recipient.user_name.length > 1 ) { #>
-						<img class="avatar" src="{{recipient.avatar}}" alt="{{recipient.user_name}}" />
+						<img class="avatar 5" src="{{{recipient.avatar}}}" alt="{{recipient.user_name}}" />
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/buddypress/common/js-templates/messages/parts/bp-messages-thread.php:44]:
```suggestion
						<img class="avatar" src="{{{recipient.avatar}}}" alt="{{recipient.user_name}}" />
```

**Review by chetansatasiya — APPROVED** [review]
@dipskakadiya can you look into my comment and see it this intenionally added or it's for a debug purpose.

**Comment by dipskakadiya** [comment]
@chetansatasiya Yeah it's debugging code by mistake pushed. I removed. Check it, please.

**Review by chetansatasiya — APPROVED** [review]
@dipskakadiya code looks good to me.

@ankitguptaindia Need to test this.

**Review by mehulkaklotar — APPROVED** [review]
@dipskakadiya This looks good to me! Tested https://prnt.sc/ulmgv2 🚀

**Comment by ExoGeek** [comment]
Hello,

This is very strange because I've updated BuddyBoss Platform and Theme to the last release and I still have the same problem.

Here is the HTML code of the gravatar : 

`Luna`

The correct URL should be : 

`https://www.gravatar.com/avatar/921149810b9d55b6be839f10eff54a06?s=150&r=g&d=identicon`

**Comment by dipskakadiya** [comment]
@ExoGeek Hey, It's not released yet. It'll be in the next release. 

Thanks for your patience.

**Comment by ExoGeek** [comment]
Okay. Sorry, I thought it was because this issue was closed. 

My Bad !

**Pull request merged**